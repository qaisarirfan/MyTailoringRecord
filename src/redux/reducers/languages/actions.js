import {REDUCERS_NAME} from '../../../utils/constants';
import {createActionName} from '../../../utils/utility';

const {languages} = REDUCERS_NAME;

export const SET_LANGUAGE = createActionName(languages, 'SET_LANGUAGE');
