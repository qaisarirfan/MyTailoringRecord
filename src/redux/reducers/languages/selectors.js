import get from 'lodash/get';
import {REDUCERS_NAME} from '../../../utils/constants';

const {languages} = REDUCERS_NAME;

export const selectLanguage = state => get(state, `${languages}.language`);
