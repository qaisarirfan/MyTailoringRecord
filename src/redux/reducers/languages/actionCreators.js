import { SET_LANGUAGE } from './actions';

export function createLanguageAction(payload) {
  return {
    type: SET_LANGUAGE,
    payload,
  };
}
