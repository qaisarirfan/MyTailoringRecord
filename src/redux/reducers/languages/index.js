import {createReducer} from '../../../utils/utility';
import {SET_LANGUAGE} from './actions';

// languages initial state
export const initialState = {
  language: 'en',
};

// languages Reducer
const reducers = {
  [SET_LANGUAGE](state, payload) {
    return {
      ...state,
      language: payload,
    };
  },
};

export default createReducer(reducers, initialState);
