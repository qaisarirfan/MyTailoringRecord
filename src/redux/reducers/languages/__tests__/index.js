import languages, {initialState} from ".."

describe("languages reducer", () => {
  it("should return the initial state", () => {
    expect(languages(undefined, {})).toEqual(initialState)
    expect(languages(undefined, {})).toMatchSnapshot()
  })
  it("should handle GET_POST_SUCCESS", () => {})
  it("should handle UPDATE_POST_SUCCESS", () => {})
  it("should handle GET_POST_FAIL", () => {})
  it("should handle GET_POST_START", () => {})
})
