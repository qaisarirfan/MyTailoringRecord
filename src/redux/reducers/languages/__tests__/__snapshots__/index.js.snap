// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`languages reducer should return the initial state 1`] = `
Object {
  "language": "en",
}
`;
